File: scaffold/src/main/java/com/vgaw/scaffold/view/MarqueeTextView.java
package com.vgaw.scaffold.view;

import android.content.Context;
import android.text.TextUtils;
import android.util.AttributeSet;

import androidx.appcompat.widget.AppCompatTextView;

/**
 * Created by xch on 2016/10/25.
 */

public class MarqueeTextView extends AppCompatTextView {
    public MarqueeTextView(Context context) {
        super(context);
        init();
    }

    public MarqueeTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public MarqueeTextView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    @Override
    public boolean isFocused() {
        return true;
    }

    private void init() {
        setEllipsize(TextUtils.TruncateAt.MARQUEE);
        setSingleLine();
    }
}

File: scaffold/src/main/java/com/vgaw/scaffold/view/MarqueeTextView.kt
package com.vgaw.scaffold.view;

import android.content.Context;
import android.text.TextUtils;
import android.util.AttributeSet;

import androidx.appcompat.widget.AppCompatTextView;

/**
 * Created by xch on 2016/10/25.
 */

class MarqueeTextView : AppCompatTextView {
    constructor(context: Context): super(context) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet): super(context, attrs) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int): super(context, attrs, defStyleAttr) {
        init()
    }

    override fun isFocused() = true

    private fun init() {
        ellipsize = TextUtils.TruncateAt.MARQUEE
        setSingleLine()
    }
}